src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11818,6 +11854,13 @@ bool isOptionalExclamationsFollowedByConditionalOperation()
                 while (this.PeekToken(index).Kind == SyntaxKind.ExclamationToken)
                     index++;
 
+                // a?.b!(
+                // a?.b![
+                // a?.b!.
+                // a?.b!?
+                //
+                // Note: for `a?.b!?`, we consume the ! as a suppression regardless of whether the ? is the start of a
+                // conditional expression or a conditional access expression.
```

Comment:
It feels like it was not a goal of this check to parse in this way. It just fell out due to the check not accounting for the details of when the `?` actually denotes another cond-access or a cond-expr. This is why I am still planning to get rid of this method in my PR #75821.

Reply:
That works for me :)